go.mod

```diff
@@ -1026,3 +1027,6 @@ exclude (
 )
 
 replace github.com/prometheus/procfs => github.com/prometheus/procfs v0.12.0
+
+// TODO: Upstream this change prior to merging the mimir.alerts.kubernetes component to main.
+replace github.com/prometheus-operator/prometheus-operator => github.com/ptodev/prometheus-operator v0.0.0-20250425071154-e2ee3c49b414
```

Comment:
I will need to merge [a PR upstream](https://github.com/prometheus-operator/prometheus-operator/pull/7486) before I merge this feature.